includes/http/WP_AI_Client_Client_Adapter.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * WordPress AI Client HTTP Client Adapter
+ *
+ * @package WordPress\AI_Client
+ * @since n.e.x.t
+ */
+
+namespace WordPress\AI_Client\HTTP;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+
+/**
+ * PSR-18 HTTP Client adapter using WordPress HTTP API
+ *
+ * This adapter allows WordPress HTTP functions to be used
+ * as a PSR-18 compliant HTTP client.
+ *
+ * @since n.e.x.t
+ */
+class WP_AI_Client_Client_Adapter implements ClientInterface {
+
+	/**
+	 * Response factory instance.
+	 *
+	 * @var ResponseFactoryInterface
+	 */
+	private $response_factory;
+
+	/**
+	 * Stream factory instance.
+	 *
+	 * @var StreamFactoryInterface
+	 */
+	private $stream_factory;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param ResponseFactoryInterface $response_factory PSR-17 Response factory.
+	 * @param StreamFactoryInterface   $stream_factory   PSR-17 Stream factory.
+	 */
+	public function __construct( ResponseFactoryInterface $response_factory, StreamFactoryInterface $stream_factory ) {
+		$this->response_factory = $response_factory;
+		$this->stream_factory   = $stream_factory;
+	}
+
+	/**
+	 * Sends a PSR-7 request and returns a PSR-7 response.
+	 *
+	 * @param RequestInterface $request The PSR-7 request.
+	 *
+	 * @return ResponseInterface The PSR-7 response.
+	 *
+	 * @throws \Exception If the WordPress HTTP request fails.
+	 */
+	public function sendRequest( RequestInterface $request ): ResponseInterface {
+		$args = $this->prepare_wp_args( $request );
+		$url  = (string) $request->getUri();
+
+		/** Ignoring PHPStan for WordPress-specific array structure. @phpstan-ignore-next-line */
+		$response = \wp_remote_request( $url, $args );
+
+		if ( \is_wp_error( $response ) ) {
+			// TODO: Update to use PHP AI Client exceptions.
+			throw new \Exception(
+				$response->get_error_message(), // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
+				$response->get_error_code() ? (int) $response->get_error_code() : 0
```

Comment:
This doesn't work in 99% of scenarios, WordPress error codes are almost always strings with text.

What I would suggest is:
* Either introduce something like a `WP_Error_Exception` that allows to maintain the WordPress error codes too, via an additional property.
* Or rely on one of the HTTP request/response specific exceptions that are being added as part of https://github.com/WordPress/php-ai-client/pull/78.

Reply:
Per the above, I think we should simply remove this line. There's no reasonable way to maintain the WordPress error code within the exception, unless we come up with a new exception class that can also store a string-shape as error code. But that's overkill IMO.